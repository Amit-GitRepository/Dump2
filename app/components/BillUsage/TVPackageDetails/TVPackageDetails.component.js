import DetailCardRow from '../BillUsage/BillUsageShared/DetailCardRow.component';
import HeaderWithCaret from '../BillUsage/BillUsageShared/HeaderWithCaret.component';
import PackageDetailHeader from '../BillUsage/BillUsageShared/PackageDetailHeader/PackageDetailHeader.component';
import PackageOverview from '../BillUsage/BillUsageShared/PackageOverview/PackageOverview.component';
import Proptypes from 'prop-types';
import React, {Component} from 'react';
import result from 'lodash/result';
import styles from './TVPackageDetails.style';
import {ICON_MAP} from '../../../config/billUsage.config';
import {translate} from '../../../language/i18n/helper';
import {WrapperCard} from '../../Shared';

class TVPackageDetails extends Component {

  render () {
    const {packageInfo, packageDetails, isConvergence, endDate, productId} = this.props;
    const type = ICON_MAP[type];
    return (
      <WrapperCard style={styles.container} contentContainerStyles={styles.contentContainerStyles}>
        <HeaderWithCaret headerColor = {styles.headerColor} customIconStyle = {styles.iconStyle} style = {styles.headerStyle}>
          <PackageDetailHeader isConvergence = {isConvergence} phoneNumber = {productId} endDate ={endDate} name = {translate('BILLS_USAGE_TRUEVISIONS')}/>
        </HeaderWithCaret>
        <PackageOverview title = {packageInfo.description} subTitle = {packageInfo.planOffer}/>
        {/* Default value = 0 , added as per discussion with design team, TODO: Remove this when API up*/}
        <DetailCardRow icon = {result(ICON_MAP, 'HDCHANNELS.titleIconPath', '')} title= {result(ICON_MAP, 'HDCHANNELS.titleText', '')} value={packageDetails.channelCountHD || 0} subText = {'Channels'}/>
        <DetailCardRow icon = {result(ICON_MAP, 'CHANNELS.titleIconPath', '')} title= {result(ICON_MAP, 'CHANNELS.titleText', '')} value={packageDetails.channelCountSD || 0} subText = {'Channels'} showSeparator={false}/>
        {/* {additionalOffers.map((Offer, index) => <DetailCardRow key = {index} rowData = {Offer}/>)}
        <ISText type = 'SEMIBOLD' style = {styles.footerText} >{'*Equipment deposite 1,000 Baht'}</ISText>
        not sure where this text was coming from hence added constant value for now */}
      </WrapperCard>
    );
  }
}
TVPackageDetails.defaultProps = {
  packageDetails: {},
  isConvergence: false,
  packageInfo: {},
  productId: '',
  endDate: 0
};
TVPackageDetails.propTypes = {
  packageDetails: Proptypes.object.isRequired,
  isConvergence: Proptypes.bool,
  packageInfo: Proptypes.object,
  productId: Proptypes.string,
  endDate: Proptypes.number
};
export default TVPackageDetails;
