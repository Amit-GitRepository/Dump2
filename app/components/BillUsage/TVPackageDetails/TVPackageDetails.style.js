import {colors, fonts} from '../../../themes/constants.styles';

export default {
  container: {
    marginVertical: 15,
    marginHorizontal: 15 // Horizonatal added separately to override thr components inherit styles
  },
  contentContainerStyles: {
    paddingBottom: 5
  },
  iconStyle: {
    marginBottom: -17
  },
  headerStyle: {
    padding: 10
  },
  footerText: {
    padding: 15,
    alignSelf: 'flex-start',
    fontSize: fonts.FONT_SIZE_SMALL,
    color: colors.PRIMARY_SUBTEXT,
    textAlign: 'right'
  },
  headerColor: colors.SECONDARY_TVISION
};
