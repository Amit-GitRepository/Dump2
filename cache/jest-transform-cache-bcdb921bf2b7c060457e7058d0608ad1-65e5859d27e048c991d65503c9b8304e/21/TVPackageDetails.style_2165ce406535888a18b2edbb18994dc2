Object.defineProperty(exports,"__esModule",{value:true});var cov_1bf2jhlxor=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/TrueVision/TVPackageDetails.style.js',hash='eb77decd317268095cf49dcc2fb15248b402e352',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/BillUsage/TrueVision/TVPackageDetails.style.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../themes/constants.styles');exports.default=

{
iconStyle:{
marginBottom:-17},

headerStyle:{
padding:10},

footerText:{
padding:15,
alignSelf:'flex-start',
fontSize:_constants.fonts.FONT_SIZE_SMALL,
color:_constants.colors.PRIMARY_SUBTEXT,
textAlign:'right'},

headerColor:_constants.colors.SECONDARY_TVISION};