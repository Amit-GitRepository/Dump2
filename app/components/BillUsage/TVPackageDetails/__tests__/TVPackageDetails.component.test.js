import React from 'react';
import renderer from 'react-test-renderer';
import TVPackageDetails from '../TVPackageDetails.component'; // Enzyme

describe('TVPackagDetail card', () => {
  const packageInfo = {
    'name': 'VGSPF01P001',
    'description': 'Sport Family HD'
  };

  const packageDetails = {
    'packageName': 'Sport Family HD',
    'price': 590,
    'channelCountHD': 19,
    'channelCountSD': 72,
    'imageURL': 'https://cms.dmpcdn.com/iservice_v2/TVS_PACKAGE/TVS_SportFamilyHD_en.png'
  };

  const productId = '123444';

  it('renders correctly', () => {
    const component = renderer.create(<TVPackageDetails packageDetails = {packageDetails} packageInfo = {packageInfo} productId={productId}/>);
    expect(component).toMatchSnapshot();
  });
  it('renders correctly when packageDetails is empty', () => {
    const component = renderer.create(<TVPackageDetails packageDetails = {{}} packageInfo = {packageInfo} productId={productId}/>);
    expect(component).toMatchSnapshot();
  });
  it('renders correctly when packageInfo is empty', () => {
    const component = renderer.create(<TVPackageDetails packageDetails = {packageDetails} packageInfo = {{}} productId={productId}/>);
    expect(component).toMatchSnapshot();
  });
});
