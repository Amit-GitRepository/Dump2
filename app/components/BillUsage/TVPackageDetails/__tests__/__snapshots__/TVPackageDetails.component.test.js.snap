// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TVPackagDetail card renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginHorizontal": 8,
      },
      Object {
        "marginHorizontal": 15,
        "marginVertical": 15,
      },
    ]
  }
>
  <View
    elevation={2}
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#B9C1C9",
          "borderRadius": 10,
          "borderWidth": 1,
          "shadowColor": "#303C46",
          "shadowOffset": Object {
            "height": 2,
            "width": 1,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 1,
        },
        Object {
          "paddingBottom": 5,
        },
      ]
    }
  >
    <View
      style={undefined}
    >
      <View>
        <View
          style={Object {}}
        >
          <View
            style={
              Array [
                Object {
                  "borderTopLeftRadius": 10,
                  "borderTopRightRadius": 10,
                  "padding": 10,
                },
                Object {
                  "padding": 10,
                },
                Object {
                  "backgroundColor": "#E82E8A",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  Array [
                    false,
                    Object {},
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#303C46",
                      },
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 14,
                          "marginBottom": -5,
                        },
                        Object {},
                      ],
                      Object {
                        "fontFamily": "SukhumvitSet-Bold",
                      },
                    ]
                  }
                >
                  TrueVisions
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#FFFFFF",
                      "fontFamily": "SukhumvitSet-Bold",
                      "fontSize": 22,
                      "marginBottom": -5,
                    }
                  }
                >
                  123444
                </Text>
              </View>
            </View>
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#E82E8A",
                "fontSize": 30,
              },
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                },
                Object {
                  "bottom": 12,
                  "marginBottom": -12,
                },
                Object {
                  "marginBottom": -17,
                },
              ],
              Object {
                "fontFamily": "iservice",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 5,
          "paddingTop": 2,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#000000",
            "fontFamily": "SukhumvitSet-Bold",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Sport Family HD
      </Text>
      <View
        style={
          Object {
            "borderBottomColor": "#E9E9EA",
            "borderBottomWidth": 1,
            "marginTop": 10,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 5,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 57,
            "justifyContent": "space-between",
            "paddingHorizontal": 5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 30,
                },
                Object {
                  "color": "#68747E",
                  "paddingRight": 10,
                },
                Object {
                  "fontFamily": "iservice",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#303C46",
                "fontFamily": "SukhumvitSet-Bold",
                "fontSize": 18,
                "marginBottom": -4,
              }
            }
          >
            HD Channel
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#303C46",
                "fontFamily": "SukhumvitSet-Text",
                "textAlign": "right",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#303C46",
                  },
                  Array [
                    Object {
                      "fontSize": 22,
                    },
                    Object {},
                  ],
                  Object {
                    "fontFamily": "SukhumvitSet-SemiBold",
                  },
                ]
              }
            >
              19
            </Text>
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                },
                Array [
                  Object {
                    "color": "#68747E",
                    "fontSize": 14,
                    "marginTop": -10,
                    "textAlign": "right",
                  },
                  Object {},
                ],
                Object {
                  "fontFamily": "SukhumvitSet-SemiBold",
                },
              ]
            }
          >
            Channels
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#E9E9EA",
            "borderBottomWidth": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 5,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 57,
            "justifyContent": "space-between",
            "paddingHorizontal": 5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 30,
                },
                Object {
                  "color": "#68747E",
                  "paddingRight": 10,
                },
                Object {
                  "fontFamily": "iservice",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#303C46",
                "fontFamily": "SukhumvitSet-Bold",
                "fontSize": 18,
                "marginBottom": -4,
              }
            }
          >
            Channel
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#303C46",
                "fontFamily": "SukhumvitSet-Text",
                "textAlign": "right",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#303C46",
                  },
                  Array [
                    Object {
                      "fontSize": 22,
                    },
                    Object {},
                  ],
                  Object {
                    "fontFamily": "SukhumvitSet-SemiBold",
                  },
                ]
              }
            >
              72
            </Text>
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                },
                Array [
                  Object {
                    "color": "#68747E",
                    "fontSize": 14,
                    "marginTop": -10,
                    "textAlign": "right",
                  },
                  Object {},
                ],
                Object {
                  "fontFamily": "SukhumvitSet-SemiBold",
                },
              ]
            }
          >
            Channels
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`TVPackagDetail card renders correctly when packageDetails is empty 1`] = `
<View
  style={
    Array [
      Object {
        "marginHorizontal": 8,
      },
      Object {
        "marginHorizontal": 15,
        "marginVertical": 15,
      },
    ]
  }
>
  <View
    elevation={2}
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#B9C1C9",
          "borderRadius": 10,
          "borderWidth": 1,
          "shadowColor": "#303C46",
          "shadowOffset": Object {
            "height": 2,
            "width": 1,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 1,
        },
        Object {
          "paddingBottom": 5,
        },
      ]
    }
  >
    <View
      style={undefined}
    >
      <View>
        <View
          style={Object {}}
        >
          <View
            style={
              Array [
                Object {
                  "borderTopLeftRadius": 10,
                  "borderTopRightRadius": 10,
                  "padding": 10,
                },
                Object {
                  "padding": 10,
                },
                Object {
                  "backgroundColor": "#E82E8A",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  Array [
                    false,
                    Object {},
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#303C46",
                      },
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 14,
                          "marginBottom": -5,
                        },
                        Object {},
                      ],
                      Object {
                        "fontFamily": "SukhumvitSet-Bold",
                      },
                    ]
                  }
                >
                  TrueVisions
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#FFFFFF",
                      "fontFamily": "SukhumvitSet-Bold",
                      "fontSize": 22,
                      "marginBottom": -5,
                    }
                  }
                >
                  123444
                </Text>
              </View>
            </View>
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#E82E8A",
                "fontSize": 30,
              },
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                },
                Object {
                  "bottom": 12,
                  "marginBottom": -12,
                },
                Object {
                  "marginBottom": -17,
                },
              ],
              Object {
                "fontFamily": "iservice",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 5,
          "paddingTop": 2,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#000000",
            "fontFamily": "SukhumvitSet-Bold",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Sport Family HD
      </Text>
      <View
        style={
          Object {
            "borderBottomColor": "#E9E9EA",
            "borderBottomWidth": 1,
            "marginTop": 10,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 5,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 57,
            "justifyContent": "space-between",
            "paddingHorizontal": 5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 30,
                },
                Object {
                  "color": "#68747E",
                  "paddingRight": 10,
                },
                Object {
                  "fontFamily": "iservice",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#303C46",
                "fontFamily": "SukhumvitSet-Bold",
                "fontSize": 18,
                "marginBottom": -4,
              }
            }
          >
            HD Channel
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#303C46",
                "fontFamily": "SukhumvitSet-Text",
                "textAlign": "right",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#303C46",
                  },
                  Array [
                    Object {
                      "fontSize": 22,
                    },
                    Object {},
                  ],
                  Object {
                    "fontFamily": "SukhumvitSet-SemiBold",
                  },
                ]
              }
            >
              0
            </Text>
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                },
                Array [
                  Object {
                    "color": "#68747E",
                    "fontSize": 14,
                    "marginTop": -10,
                    "textAlign": "right",
                  },
                  Object {},
                ],
                Object {
                  "fontFamily": "SukhumvitSet-SemiBold",
                },
              ]
            }
          >
            Channels
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#E9E9EA",
            "borderBottomWidth": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 5,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 57,
            "justifyContent": "space-between",
            "paddingHorizontal": 5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 30,
                },
                Object {
                  "color": "#68747E",
                  "paddingRight": 10,
                },
                Object {
                  "fontFamily": "iservice",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#303C46",
                "fontFamily": "SukhumvitSet-Bold",
                "fontSize": 18,
                "marginBottom": -4,
              }
            }
          >
            Channel
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#303C46",
                "fontFamily": "SukhumvitSet-Text",
                "textAlign": "right",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#303C46",
                  },
                  Array [
                    Object {
                      "fontSize": 22,
                    },
                    Object {},
                  ],
                  Object {
                    "fontFamily": "SukhumvitSet-SemiBold",
                  },
                ]
              }
            >
              0
            </Text>
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                },
                Array [
                  Object {
                    "color": "#68747E",
                    "fontSize": 14,
                    "marginTop": -10,
                    "textAlign": "right",
                  },
                  Object {},
                ],
                Object {
                  "fontFamily": "SukhumvitSet-SemiBold",
                },
              ]
            }
          >
            Channels
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`TVPackagDetail card renders correctly when packageInfo is empty 1`] = `
<View
  style={
    Array [
      Object {
        "marginHorizontal": 8,
      },
      Object {
        "marginHorizontal": 15,
        "marginVertical": 15,
      },
    ]
  }
>
  <View
    elevation={2}
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#B9C1C9",
          "borderRadius": 10,
          "borderWidth": 1,
          "shadowColor": "#303C46",
          "shadowOffset": Object {
            "height": 2,
            "width": 1,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 1,
        },
        Object {
          "paddingBottom": 5,
        },
      ]
    }
  >
    <View
      style={undefined}
    >
      <View>
        <View
          style={Object {}}
        >
          <View
            style={
              Array [
                Object {
                  "borderTopLeftRadius": 10,
                  "borderTopRightRadius": 10,
                  "padding": 10,
                },
                Object {
                  "padding": 10,
                },
                Object {
                  "backgroundColor": "#E82E8A",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  Array [
                    false,
                    Object {},
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#303C46",
                      },
                      Array [
                        Object {
                          "color": "#FFFFFF",
                          "fontSize": 14,
                          "marginBottom": -5,
                        },
                        Object {},
                      ],
                      Object {
                        "fontFamily": "SukhumvitSet-Bold",
                      },
                    ]
                  }
                >
                  TrueVisions
                </Text>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#FFFFFF",
                      "fontFamily": "SukhumvitSet-Bold",
                      "fontSize": 22,
                      "marginBottom": -5,
                    }
                  }
                >
                  123444
                </Text>
              </View>
            </View>
          </View>
        </View>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#E82E8A",
                "fontSize": 30,
              },
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "transparent",
                },
                Object {
                  "bottom": 12,
                  "marginBottom": -12,
                },
                Object {
                  "marginBottom": -17,
                },
              ],
              Object {
                "fontFamily": "iservice",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 5,
          "paddingTop": 2,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#000000",
            "fontFamily": "SukhumvitSet-Bold",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        
      </Text>
      <View
        style={
          Object {
            "borderBottomColor": "#E9E9EA",
            "borderBottomWidth": 1,
            "marginTop": 10,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 5,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 57,
            "justifyContent": "space-between",
            "paddingHorizontal": 5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 30,
                },
                Object {
                  "color": "#68747E",
                  "paddingRight": 10,
                },
                Object {
                  "fontFamily": "iservice",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#303C46",
                "fontFamily": "SukhumvitSet-Bold",
                "fontSize": 18,
                "marginBottom": -4,
              }
            }
          >
            HD Channel
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#303C46",
                "fontFamily": "SukhumvitSet-Text",
                "textAlign": "right",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#303C46",
                  },
                  Array [
                    Object {
                      "fontSize": 22,
                    },
                    Object {},
                  ],
                  Object {
                    "fontFamily": "SukhumvitSet-SemiBold",
                  },
                ]
              }
            >
              19
            </Text>
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                },
                Array [
                  Object {
                    "color": "#68747E",
                    "fontSize": 14,
                    "marginTop": -10,
                    "textAlign": "right",
                  },
                  Object {},
                ],
                Object {
                  "fontFamily": "SukhumvitSet-SemiBold",
                },
              ]
            }
          >
            Channels
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomColor": "#E9E9EA",
            "borderBottomWidth": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 5,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 57,
            "justifyContent": "space-between",
            "paddingHorizontal": 5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 30,
                },
                Object {
                  "color": "#68747E",
                  "paddingRight": 10,
                },
                Object {
                  "fontFamily": "iservice",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#303C46",
                "fontFamily": "SukhumvitSet-Bold",
                "fontSize": 18,
                "marginBottom": -4,
              }
            }
          >
            Channel
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#303C46",
                "fontFamily": "SukhumvitSet-Text",
                "textAlign": "right",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#303C46",
                  },
                  Array [
                    Object {
                      "fontSize": 22,
                    },
                    Object {},
                  ],
                  Object {
                    "fontFamily": "SukhumvitSet-SemiBold",
                  },
                ]
              }
            >
              72
            </Text>
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                },
                Array [
                  Object {
                    "color": "#68747E",
                    "fontSize": 14,
                    "marginTop": -10,
                    "textAlign": "right",
                  },
                  Object {},
                ],
                Object {
                  "fontFamily": "SukhumvitSet-SemiBold",
                },
              ]
            }
          >
            Channels
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
